### ISSUE TITLE
BBC: Blead Breaks Scope::Upper

### ISSUE THREAD

**Issue by cjg-cguevara** [Perl/perl5#19212]
This is a bug report for perl from "Carlos Guevara" <carlos@carlosguevara.com>,
generated with the help of perlbug 1.42 running under perl 5.35.5.

-----------------------------------------------------------------
[Please describe your issue here]

BBC: Blead Breaks Scope::Upper

Please see http://matrix.cpantesters.org/?dist=Scope-Upper

[Please do not change anything below this line]
-----------------------------------------------------------------
---
Flags:
    category=core
    severity=low
---
Site configuration information for perl 5.35.5:

Configured by cpan at Thu Oct 21 15:53:57 EDT 2021.

Summary of my perl5 (revision 5 version 35 subversion 5) configuration:
  Commit id: a333292f05e3fc0ed682ccc8c12c33a4c9d7aafd
  Platform:
    osname=freebsd
    osvers=13.0-release-p4
    archname=amd64-freebsd-thread-multi
    uname='freebsd cjg-freebsd13 13.0-release-p4 freebsd 13.0-release-p4 #0: tue aug 24 07:33:27 utc 2021 root@amd64-builder.daemonology.net:usrobjusrsrcamd64.amd64sysgeneric amd64 '
    config_args='-des -Dprefix=~/bin/perl-blead -Dscriptdir=~/bin/perl-blead/bin -Dusedevel -Duse64bitall -Duseithreads'
    hint=recommended
    useposix=true
    d_sigaction=define
    useithreads=define
    usemultiplicity=define
    use64bitint=define
    use64bitall=define
    uselongdouble=undef
    usemymalloc=n
    default_inc_excludes_dot=define
  Compiler:
    cc='cc'
    ccflags ='-DHAS_FPSETMASK -DHAS_FLOATINGPOINT_H -fno-strict-aliasing -pipe -fstack-protector-strong -I/usr/local/include'
    optimize='-O2'
    cppflags='-DHAS_FPSETMASK -DHAS_FLOATINGPOINT_H -fno-strict-aliasing -pipe -fstack-protector-strong -I/usr/local/include'
    ccversion=''
    gccversion='FreeBSD Clang 11.0.1 (git@github.com:llvm/llvm-project.git llvmorg-11.0.1-0-g43ff75f2c3fe)'
    gccosandvers=''
    intsize=4
    longsize=8
    ptrsize=8
    doublesize=8
    byteorder=12345678
    doublekind=3
    d_longlong=define
    longlongsize=8
    d_longdbl=define
    longdblsize=16
    longdblkind=3
    ivtype='long'
    ivsize=8
    nvtype='double'
    nvsize=8
    Off_t='off_t'
    lseeksize=8
    alignbytes=8
    prototype=define
  Linker and Libraries:
    ld='cc'
    ldflags ='-pthread -Wl,-E  -fstack-protector-strong -L/usr/local/lib'
    libpth=/usr/lib /usr/local/lib /usr/lib/clang/11.0.1/lib
    libs=-lpthread -ldl -lm -lcrypt -lutil
    perllibs=-lpthread -ldl -lm -lcrypt -lutil
    libc=
    so=so
    useshrplib=false
    libperl=libperl.a
    gnulibc_version=''
  Dynamic Linking:
    dlsrc=dl_dlopen.xs
    dlext=so
    d_dlsymun=undef
    ccdlflags=' '
    cccdlflags='-DPIC -fPIC'
    lddlflags='-shared	-L/usr/local/lib -fstack-protector-strong'

---
@INC for perl 5.35.5:
    /home/cpan/bin/perl-blead/lib/site_perl/5.35.5/amd64-freebsd-thread-multi
    /home/cpan/bin/perl-blead/lib/site_perl/5.35.5
    /home/cpan/bin/perl-blead/lib/5.35.5/amd64-freebsd-thread-multi
    /home/cpan/bin/perl-blead/lib/5.35.5

---
Environment for perl 5.35.5:
    HOME=/home/cpan
    LANG=C.UTF-8
    LANGUAGE (unset)
    LC_ALL=C
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=/home/cpan/bin/perl-blead/bin:/home/cpan/bin:/sbin:/bin:/usr/sbin:/usr/bin:/usr/local/sbin:/usr/local/bin:/home/cpan/bin
    PERL_BADLANG (unset)
    SHELL=/usr/local/bin/bash

**Comment by hvds** [comment]
This bisects to 3b54923c12 - fyi @nwc10 @vpit

**Comment by jkeenan** [comment]
> This bisects to [3b54923](https://github.com/Perl/perl5/commit/3b54923c12732530d17e87f548066ceadc63daa4) - fyi @nwc10 @vpit

Hugo beat me to it by 5 minutes!

Bisecting with the following invocation:
```
perl Porting/bisect.pl --start=1707eb284b1f9646240f4f4ca393f6fa309b51fc \
--end=7ab72683be77c37dbd4c1d1194eccdc905d8ff88 \
--module=Scope::Upper
```
I found:
```
3b54923c12732530d17e87f548066ceadc63daa4 is the first bad commit
commit 3b54923c12732530d17e87f548066ceadc63daa4
Author:     Nicholas Clark <nick@ccl4.org>
AuthorDate: Mon Jun 21 19:34:29 2021 +0000
Commit:     Ricardo Signes <rjbs@semiotic.systems>
CommitDate: Fri Oct 15 09:28:27 2021 -0400

    Add a new warning experimental::for_list.

 lib/warnings.pm   | 15 ++++++++++-----
 regen/warnings.pl |  4 +++-
 warnings.h        |  2 ++
 3 files changed, 15 insertions(+), 6 deletions(-)
bisect run success
That took 1294 seconds.
```

**Comment by jkeenan** [comment]
Reported upstream at [cpan 139823](https://rt.cpan.org/Ticket/Display.html?id=139823).

**Comment by jkeenan** [comment]
We can explore this BBC problem in [this GH repository](https://github.com/jkeenan/p5-Scope-Upper/tree/jkeenan/gh-19192), created with permission of upstream maintainer Vincent Pit.

In that repo, branch `main` is Scope-Upper-0.32 from CPAN plus a `.gitignore`.  The problem which we're trying to address is a failure in `t/07-context_info.t` that appears in `perl-5.35.5` and forward.  As this involves XS code, warnings, `caller()` and a customized testing setup, it will benefit from many eyeballs.

Thank you very much.
Jim Keenan

**Comment by tonycoz** [comment]
This looks like the test is simply fragile, I've commented on the upstream ticket.

**Comment by jkeenan** [comment]
> Reported upstream at [cpan 139823](https://rt.cpan.org/Ticket/Display.html?id=139823).

A patch has been submitted to the upstream maintainer in that RT ticket.

**Comment by jkeenan** [comment]
> > Reported upstream at [cpan 139823](https://rt.cpan.org/Ticket/Display.html?id=139823).
> 
> A patch has been submitted to the upstream maintainer in that RT ticket.

Maintainer Vincent Pit has accepted the patch and released [Scope-Upper-0.33](https://metacpan.org/dist/Scope-Upper) to CPAN, so we can now close this ticket.
